src/frontend/js/modules/avatar.js

```diff
@@ -574,6 +574,91 @@
       function displayUserStatus() {
         return self.objectType === 'user' ? !!self.avatar.id && !self.hideUserStatus : false;
       }
+    })
+    .factory('EsnInitialsAvatarGeneratorService', function() {
+      return {
+        generate
+      };
+
+      /**
+       * generate an avatar from username or email
+       *
+       * @param {String} username - the username or email
+       * @returns {String} - the generated avatar string
+       */
+      function generate(username) {
+        const text = generateInitials(username);
+        const color = generateColor(username);
+
+        return drawAvatar(text, color);
+      }
+
+      /**
+       * generate initials from username or email
+       *
+       * @param {String} name - the username or email
+       * @returns {String} - the initials
+       */
+      function generateInitials(name) {
+        const initials = name.split(' ').map(c => c[0].toUpperCase());
+
+        return initials.join('');
+      }
+
+      /**
+       * generate color from username or email
+       *
+       * @param {String} name - the username or email
+       * @returns {String} - the color
+       */
+      function generateColor(seed) {
+        let hash = 0;
+
+        seed.split('').forEach(c => {
+          // eslint-disable-next-line no-bitwise
+          hash = ((hash << 5) - hash) + c.charCodeAt(0);
+        });
+
+        // eslint-disable-next-line no-bitwise
+        const color = (hash & 0x00FFFFFF)
+          .toString(16)
+          .toUpperCase();
+
+        return '#' + ('000000' + color).slice(-6);
+      }
+
+      /**
+       * generates a base64 string canvas element with the given text and color
+       *
+       * @param {String} text - the text to draw
+       * @param {String} color - the background color
+       * @returns {String} - the generated canvas element
+       */
+      function drawAvatar(text, color) {
+        const canvas = document.createElement('canvas');
+        const context = canvas.getContext('2d');
+
+        canvas.width = 200;
+        canvas.height = 200;
+
+        context.fillStyle = color;
+        context.beginPath();
+        context.ellipse(
+          canvas.width / 2, canvas.height / 2,
+          canvas.width / 2, canvas.height / 2,
+          0,
+          0, Math.PI * 2
+        );
+        context.fill();
+
+        context.font = (canvas.height / 3) + 'px serif';
+        context.fillStyle = '#000';
+        context.textAlign = 'center';
+        context.textBaseline = 'middle';
+        context.fillText(text, canvas.width / 2, canvas.height / 2);
+
+        return canvas.toDataURL();
```

Comment:
Maybe we can reduce image quality, this could perhaps improve the loading  ?
`canvas.toDataURL('image/jpeg', 0.5);`

Reply:
we don't load anything from the network or whatsoever, it's just a base64 string generated from a service that exists within the frontend app itself.